### PR TITLE
add triples.py file in the physics

### PR DESCRIPTION
**Pull request by sylviemonet**
**Context:** Add a new file triples.py in the physics module to include all triples in Bargmann representations of the states and transformations.
This PR will set up the basic physics for Bargmann objects.

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: mrmustard/physics/triples.py [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+
+def displaced_squeezed_vacuum_state_ABC_triples(
+    x: Union[Scalar, Vector],
+    y: Union[Scalar, Vector],
+    r: Union[Scalar, Vector],
+    phi: Union[Scalar, Vector],
+) -> Union[Matrix, Vector, Scalar]:
+    r"""Returns the ABC triples of a displaced squeezed vacuum state.
+
+    ValueError will be raise if the dimensions of ``x``, ``y``, ``r`` and ``phi`` are different.
```
[the remark this answers is not in the document]
Comment (SamFerracin) [reply mrmustard/physics/triples.py:107]: more than ``Vector``, I would use ``Sequence``

File: mrmustard/physics/triples.py [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+#  ~~~~~~~~~~~~
+#  Pure States
+#  ~~~~~~~~~~~~
+
+
+def vacuum_state_ABC_triples(num_modes: int) -> Union[Matrix, Vector, Scalar]:
+    r"""Returns the ABC triples of the pure vacuum state.
+
+    Args:
+        num_modes (int): number of modes
```
[the remark this answers is not in the document]
Comment (sylviemonet) [reply mrmustard/physics/triples.py]: Did by replacement for all.

File: mrmustard/physics/triples.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sylviemonet) [reply mrmustard/physics/triples.py:133]: Vector goes to list.

**Comment by SamFerracin** [comment]
This looks good to me now. Take a look and merge whenever you feel it's ready :)
Question:
- Are we missing some tests? Especially on those functions that do something non-trivial to the input parameters to change their lengths, or on those that do the `math.block` thing

**Review by SamFerracin — APPROVED** [review]
This looks good to me now. Take a look and merge whenever you feel it's ready :)
Question:

    Are we missing some tests? Especially on those functions that do something non-trivial to the input parameters to change their lengths, or on those that do the math.block thing

File: doc/code/physics.rst [hunk 3]
```diff
@@ -5,8 +5,7 @@ mrmustard.physics
     :maxdepth: 1
 
     physics/ansatze
-    physics/bargmann
```
Comment (ziofil) [inline doc/code/physics.rst:8]: was this indended? (removing these folders from the docs)

File: pyproject.toml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "mrmustard"
-version = "0.7.0"
+version = "0.7.1"
```
Comment (ziofil) [inline pyproject.toml:3]: why?

File: mrmustard/physics/triples.py [hunk 5]
```diff
@@ -0,0 +1,460 @@
...
+
+import numpy as np
+
+
+#  ~~~~~~~~~
+#  Utilities
+#  ~~~~~~~~~
+
+
+def _X_matrix_for_unitary(n_modes: int) -> Matrix:
```
Comment (ziofil) [inline mrmustard/physics/triples.py:32]: why not defining it statically? this way it needs to be recreated over and over

Comment (ziofil) [reply mrmustard/physics/triples.py:32]: oh you need it as a function of `n_modes`

File: mrmustard/physics/triples.py [hunk 6]
```diff
@@ -0,0 +1,460 @@
...
+    c = math.prod(1 / math.sqrt(math.cosh(r)))
+
+    return A, b, c
+
+
+def beamsplitter_gate_Abc(
+    theta: Union[Scalar, Iterable], phi: Union[Scalar, Iterable] = 0
+) -> Union[Matrix, Vector, Scalar]:
+    r"""
+    The ``(A, b, c)`` triple of a tensor product of two-mode beamsplitter gates.
```
Comment (ziofil) [inline mrmustard/physics/triples.py:319]: you mean `n` bs gates?

**Review by ziofil — APPROVED** [review]
Very nice 👍
For Q1 we will also need the Abc of the basis change kernels: up to you to add them to this PR or to open a separate one.

**Pull request merged**